src/app/services/organization-redirect-guard.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from "@angular/core";
+import { ActivatedRouteSnapshot, CanActivate, Router } from "@angular/router";
+
+import { I18nService } from "jslib-common/abstractions/i18n.service";
+import { OrganizationService } from "jslib-common/abstractions/organization.service";
+import { PlatformUtilsService } from "jslib-common/abstractions/platformUtils.service";
+import { Permissions } from "jslib-common/enums/permissions";
+import { Utils } from "jslib-common/misc/utils";
+
+@Injectable()
+export class OrganizationRedirectGuardService implements CanActivate {
+  constructor(
+    private router: Router,
+    private platformUtilsService: PlatformUtilsService,
+    private i18nService: I18nService,
+    private organizationService: OrganizationService
+  ) {}
+
+  async canActivate(route: ActivatedRouteSnapshot) {
+    const orgs = await this.organizationService.getAll();
+    const permissions = route.data == null ? null : (route.data.permissions as Permissions[]);
+
+    const allowedOrgs = orgs
+      .filter((org) => org.hasAnyPermission(permissions))
+      .sort(Utils.getSortFunction(this.i18nService, "name"));
```

Comment:
We're reusing the same logic here as in the navbar. What do you think of moving it to say the organization service, with a simpler API?

`organizationService.canViewOrganizationPage()` ... or something.

Reply:
I don't want it in jslib (or the very bloated `OrganizationService`) because it's specific to just one client. I agree with your sentiment though, I've created a `PermissionsService` which is a wrapper around this. This is probably a better interface than dealing with the `organizationRoutePermissions` object directly.